src/smart_module/smart_module.py

```diff
@@ -119,17 +154,29 @@ def discover(self):
             print("Real Smart Module hosting asset ", self.asset.id, self.asset.type,
                   self.asset.context)
 
-        self.log.info("Performing Discovery...")
-        subprocess.call("sudo ./host-hapi.sh", shell=True)
+        attempt = 1
+        while self.comm.broker_name is None and attempt <= 5: # or?
+            self.log.info("Performing Discovery...")
+            self.find_broker()
+            self.log.info("Waiting Broker information on attempt: %d.", attempt)
+            if self.comm.broker_name is None:
```

Comment:
We might need to set a sleep here to avoid a race condition on finding the MQTT broker and becoming the Broker.